### PR TITLE
fix(atomic): automatic facets not showing in refine modal when no static facets

### PR DESCRIPTION
**Pull request by ghost**
https://coveord.atlassian.net/browse/KIT-2665

Current behavior:
Automatic facets don't show in refine modal unless there is at least 1 static facet.

Expected behavior:
Automatic facets should show no matter what.

### CODE REVIEW

File: packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx [hunk 1]
```diff
@@ -187,12 +187,6 @@ export class AtomicRefineModal implements InitializableComponent {
   }
 
   private renderFilters() {
-    const isFacetElements = this.bindings.store.getFacetElements().length > 0;
-
-    if (!isFacetElements) {
-      return;
-    }
-
```
Comment (aprudhomme-coveo) [inline packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx:195]: I think this was added just to improve performance. Everything works fine without it.

Comment (aprudhomme-coveo) [reply packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx:195]: ah yes, you are right ! I completely missed that 😅

File: packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx [hunk 2]
```diff
@@ -188,8 +188,10 @@ export class AtomicRefineModal implements InitializableComponent {
 
   private renderFilters() {
     const isFacetElements = this.bindings.store.getFacetElements().length > 0;
+    const isAutomaticFacets =
+      this.bindings.engine.state.automaticFacetSet?.set !== undefined;
 
-    if (!isFacetElements) {
+    if (!isFacetElements && !isAutomaticFacets) {
```
Comment (mrrajamanickam-coveo) [inline packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx:194]: Wonder why this wasn't caught by tests? Any test gaps that can be fixed?

Comment (btaillon) [inline packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx:194]: Don't really like the word `is` here, which usually refers to one thing.

Maybe `hasFacetElements` and `hasAutomaticFacets`?

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx (hunk 2 above)
Comment (alexprudhomme) [reply packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx]: I added corresponding tests !

**Pull request merged**